build_alpine.Dockerfile

```
@@ -0,0 +1,60 @@
+# syntax = docker/dockerfile:1.2
+
+# Builds a light albatross image from sources.
+# Requires Docker buildkit.
+
+ARG RUST_IMAGE=rust:1-alpine
+FROM $RUST_IMAGE AS builder
+
+# Switch to Rust nightly
+RUN rustup update nightly; rustup default nightly;
+
+# Fetch dependencies.
+RUN apk update && apk add musl-dev libressl-dev protoc
+
+# Copy sources.
+COPY . /build
+WORKDIR /build
+
+# Build.
+RUN \
+  --mount=type=cache,target=/build/target \
+  # See https://doc.rust-lang.org/cargo/guide/cargo-home.html
+  --mount=type=cache,target=/usr/local/cargo/registry \
+  --mount=type=cache,target=/usr/local/cargo/git \
+  cargo build --bin nimiq-client && cp /build/target/debug/nimiq-client /build/
```

```suggestion
  cargo build --bin nimiq-client --release && cp /build/target/release/nimiq-client /build/
```
Would it be fair to create a release build?

---

@jeffesquivels I remember how in the now dismissed commits where you changed the dockerfiles, you had a `BUILD` env variable here to set debug or release build. Do you think this is still a good idea? If not set, it would default to debug.

---

@sisou That would be ideal. Using the ARG instruction allows for `docker build . --build-arg ENV=release`.
https://docs.docker.com/engine/reference/builder/#arg

Something like
```dockerfile
ARG ENV=debug
RUN cargo build --profile $ENV
```